GeosCore/mercury_mod.F90

```diff
@@ -692,38 +724,78 @@ SUBROUTINE ChemMercury( Input_Opt,  State_Chm, State_Diag,                 &
        !--------------------------------------------------------------------
        IF ( IERR < 0 ) THEN
 
+#ifdef MODEL_GEOS
+          ! Save a backup copy of C (for GEOS-Chem in GEOS only)
+          localC    = C
+#endif
+
           ! Reset first time step and start concentrations
           ! Retry the integration with non-optimized
           ! settings
-          RCNTRL(3)  = 0e+0_fp
-          CALL Init_KPP( )
-          VAR = C(1:NVAR)
-          FIX = C(NVAR+1:NSPEC)
+          RCNTRL(3) = 0e+0_fp
+          C         = 0.0_dp
+
```

Comment:
VAR and FIX are now local pointers within the KPP integrator file. This was necessary in order to remove the non-threadsafe EQUIVALENCE statements in gckpp_Global.F90